packages/block-editor/src/components/inserter/menu.js

```diff
@@ -177,83 +178,81 @@ function InserterMenu(
 		isAppender,
 	] );
 
-	const blocksTab = useMemo(
-		() => (
+	const blocksTab = useMemo( () => {
+		if ( selectedTab !== 'blocks' ) {
```

Comment:
I think this will make the tabs less performant, because we'll need to re-run this code every time the selected tab changes. It would be better to execute the code once and have it cached until we need to change it for another reason.